src/jdk.incubator.vector/share/classes/jdk/incubator/vector/AbstractMask.java

```diff
@@ -377,8 +383,13 @@ VectorMask<E> checkIndex0(long offset, long length,
             // when offset is Integer.MIN_VALUE.
             // 0 <= firstGoodIndex <= vlength
             int firstGoodIndex = (int) -Math.max(offset, -vlength);
-            VectorMask<E> badMask2 =
+            if (vectorSpecies().elementType() == Halffloat.class) {
+                badMask2 =
+                    iota.compare(LT, iota.broadcast(Float.floatToFloat16((float)firstGoodIndex)));
```

Comment:
There is a flavor of  compare API which compares against scalar value, it can be used directly.